### DISCUSSION TITLE
Dynamic the number of column table

### DISCUSSION THREAD

**Discussion by DuyThaiNgx** [Budibase/budibase#14963] [Help]
I have a table data from a API data source: http://localhost:8080/service/api/data/history/{{id}}.
This api response many fields, i can transformer data.
I want to display a field as a column in tables, but each _id_ has different field means that when i want to display field data on table, each table will have a different number of columns.
So how can i create a dynamic tables.
Example: when i call into http://localhost:8080/service/api/data/history/59, service 59 with have 5 field, then i can display service's data table with 5 cols.
But when i call http://localhost:8080/service/api/data/history/58, service 58 just has 3 field, i want to display service's data table with 3 cols.
And each service id has different name field.

**Comment by ConorWebb96** [comment]
Hey @DuyThaiNgx,

Budibase isn't suited for this use case. 

My best advice would be to find a row with the majority of the fields. Then, send a REST request in the builder that will return that row. This will build the schema that Budibase uses to generate fields/columns. This way, you'll have the majority, if not all, of the columns used, and therefore, the tables will display the majority of the rows.

You may be able to perform some tricks by using a data provider as the source of a table and passing the data from a different data provider into the previous one that is set to JSON/CSV type. However, this will limit the table's functionality.

I hope this helps.
